lib/models/settings.dart

```diff
@@ -13,36 +13,38 @@ part 'settings.enums.dart';
 // Root settings //
 // ///////////// //
 
-@freezed
+@Freezed(fromJson: true, toJson: true)
 class Settings with _$Settings implements TomlEncodableValue {
   
   const Settings._();
 
   const factory Settings({
-    required int captureDelaySeconds,
-    required bool displayConfetti,
-    required double collageAspectRatio,
-    required double collagePadding,
-    required bool singlePhotoIsCollage,
-    required String templatesFolder,
-    required HardwareSettings hardware,
-    required OutputSettings output,
+    @Default(5) int captureDelaySeconds,
+    @Default(1.5) double collageAspectRatio,
+    @Default(0) double collagePadding,
+    @Default(true) bool displayConfetti,
+    @Default(true) bool singlePhotoIsCollage,
+    @Default("") String templatesFolder,
+    @Default(HardwareSettings()) HardwareSettings hardware,
+    @Default(OutputSettings()) OutputSettings output,
   }) = _Settings;
 
-  factory Settings.withDefaults() {
-    return Settings(
-      captureDelaySeconds: 5,
-      collageAspectRatio: 1.5,
-      collagePadding: 0,
-      displayConfetti: true,
-      singlePhotoIsCollage: true,
-      templatesFolder: _getHome(),
-      hardware: HardwareSettings.withDefaults(),
-      output: OutputSettings.withDefaults(),
-    );
+  factory Settings.withDefaults() => Settings.fromJson({});
+
+  factory Settings.fromJson(Map<String, Object?> json) {
+    // Default settings
+    if (!json.containsKey("templatesFolder")) {
+      json["templatesFolder"] = _getHome();
+    }
+    if (!json.containsKey("hardware")) {
+      json["hardware"] = HardwareSettings.withDefaults().toJson();
+    }
+    if (!json.containsKey("output")) {
+      json["output"] = OutputSettings.withDefaults().toJson();
+    }
+
+    return _$SettingsFromJson(json);
```

Comment:
This is the magical part huh?

Reply:
Yes, this is my newly proposed of adding defaults. This gives us some advantages:
- It works when there is not settings file at all and a new object needs to be constructed (while not all defaults are `const`)
- It also works when settings keys were missing from the config file (at any level)
- There is no redundancy, this is the only place for the non-const defaults